src/Arcus.EventGrid/Contracts/CloudEvent.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using Arcus.EventGrid.Contracts.Interfaces;
+using GuardNet;
+using Microsoft.Azure.EventGrid.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Arcus.EventGrid.Contracts
+{
+    public class CloudEvent<TData> : IEvent
```

Comment:
We should try and see if we can plug in the official SDK rather than maintaining our own - https://github.com/cloudevents/sdk-csharp